cmd/erasure-object.go

```diff
@@ -818,7 +819,9 @@ func (er erasureObjects) deleteObject(ctx context.Context, bucket, object string
 			if disks[index] == nil {
 				return errDiskNotFound
 			}
-			err := cleanupDir(ctx, disks[index], minioMetaTmpBucket, tmpObj)
+			tctx, cancel := context.WithTimeout(ctx, 5*time.Second)
```

Comment:
You still need to use context.Background() here.

Basically, remote storage requests can get canceled because they are http requests (and not like posix). cleanupDir() can return with context canceled here without cleaning up data in remote disks.

Reply:
This is intentional so that they don't pile up @vadmeste we have another routine to pick up stale entries every 24hrs

The intention here is that we have a deadline for deletes. So we don't pile it up on the server.